src/molotov.js

```diff
@@ -0,0 +1,328 @@
+   * from passed cocktail classes. And unifying
+   * their interface. After this our supers will be
+   * ready to combine with plugins for *REAL* mixins.
+   *
+   * @returns {SuperMixologist}
+   *   Returns supers with overrides from cocktail classes.
+   *
+   */
+  mixSupers(): SuperMixologist {
+    const superMixologist = new SuperMixologist(this);
+    superMixologist.resolve();
+    return superMixologist;
+  }
+
+  /**
+   * Mixes a molotov cocktail
+   *   for this molotov. The PolttoPullo class
+   *   is ready to be resolved incorporating
+   *   Cocktail classes and overriden plugin
+   *   composition.
+   *
+   * @returns {Polttopullo}
+   *   A mixed PolttoPullo plugins class ready
+   *   to be resolved (polttopull.resolve().then((res) => // use away))
+   */
+  getMolotov(): Polttopullo {
+    const resolvedSupers = this.mixSupers();
+    return new Polttopullo(resolvedSupers.molotov);
+  }
+
+  /**
+   * Sets the namespace for this molotov use.
+   *
+   * @param {string} nameSpace
+   *   The namespace of the module implementing molotov.
+   *   This will be used to interpret configuration passed
+   *   in molotov config.
+   *
+   * @returns {void}
+   */
+  setNameSpace(nameSpace: string): void {
+    this.molotovNameSpace = nameSpace;
+  }
+
+  /**
+   * Get the namespace string for this molotov implementation.
+   *
+   * @returns {string}
+   *   The namespace of this molotov implementation.
+   */
+  getNameSpace(): string {
+    return this.molotovNameSpace;
+  }
+
+  /**
+   * Sets any config overrides for this molotov implementation.
+   *   Usually reserved for any dynamic plugin compositions or
+   *   for modules implementing a molotov implementer with
+   *   plugin composition overrides.
+   *
+   * @param {(overrideConfig | {})} overrides
+   *   Configuration overrides for this molotov instance. These will
+   *   override existing configuration in the passed in molotovConfig.
+   *
+   * @returns {void}
+   */
+  setConfigOverrides(overrides: (overrideConfig | {})): void {
+    this.configOverrides = overrides;
+  }
+
+  /**
+   * Gets configuration overrides for this molotov.
+   *
+   * @returns {(overrideConfig | {})}
+   *   Configuration overrides for this molotov instance. These will
+   *   override existing configuration in the passed in molotovConfig.
+   */
+  getConfigOverrides(): (overrideConfig | {}) {
+    return this.configOverrides;
+  }
+
+  /**
+   * Sets the array of Cocktail classes for this molotov.
+   *
+   * @param {((Cocktail[] | []))} cocktails
+   *   An array of cocktail classes to be used by modules
+   *   utilizing molotov implementing modules to override
+   *   or provide additional supers, mixins, and plugin
+   *   definitions.
+   *
+   * @returns {void}
+   */
+  setCocktails(cocktails: Array<?Cocktail>): void {
+    this.cocktails = cocktails;
+  }
+  /**
+   * Gets the Cocktail array for this molotov.
+   * It's martini time!
```

Comment:
![pretend this is a martini and not champagne](https://media.giphy.com/media/GCLlQnV7wzKLu/giphy.gif)